pilot/pkg/bootstrap/server.go

```diff
@@ -740,6 +743,13 @@ func (s *Server) initEventHandlers() error {
 				Reason: []model.TriggerReason{model.ConfigUpdate},
 			}
 			s.EnvoyXdsServer.ConfigUpdate(pushReq)
+			if s.statusReporter != nil {
```

Comment:
This is a pretty dangerous place to add this - for EDS we are bypassing all the Event and go straight to the real interface. 

Pretty new code to hook the components - it has changed many times and unlikely to stay stable.

Reply:
Is there a more stable place for this logic?

Reply:
For example in the k8s package - since this is specific to K8S anyways. So you won't even need to have much exposed outside of the package, if you also move the rest of the logic to that package. All self-contained, no need for model changes or to expose it here. 

If you need - you can define a new interface for writing the status update - implemented in the kube package.